File: 2534/2534.py
# bee 2534 - Exame Geral
# Alonso Martins
# 07/08/2023
while True:
    try:
        N, Q = input().split()
        notas = []
        for i in range(int(N)):
            notas.append(int(input()))
        notas.sort(reverse = True)
        for i in range(int(Q)):
            print(notas[int(input())-1])
    except EOFError:
        break

File: 2534/2534.c
// bee 2534 - Exame Geral
// Alonso Martins
// 08/04/2024
#include <stdio.h>
#include <stdlib.h>

int comparar(const void *a, const void *b)
{
    return *(int *)b - *(int *)a;
}
int main()
{
    int N, Q, i, c;
    int notas[100];

    while (scanf("%d %d", &N, &Q) != EOF)
    {
        for (i = 0; i < N; i ++)
            scanf("%d", &notas[i]);

        qsort(notas, N, sizeof(int), comparar);

        for (i = 0; i < Q; i ++)
        {
            scanf("%d", &c);
            printf("%d\n", notas[c - 1]);
        }
    }
    return 0;
}
